src/lib/server/post.js

```diff
@@ -4,50 +4,90 @@ import { getBackendClient } from '$lib/server/client'
 
 const debug = createDebugMessages('APP:$lib/server/post')
 
-const POST_QUERY = `{
-  post(filter: { slug: { _eq: "{{slug}}" } }) {
-      slug
-      status
-      type
-      title
-      body
-      publish_on
-      auto_publish
-      thumbnail {
-          id
-          filename_disk
-          filename_download
-      }
-  }
-}`
+export default class Post {
+    /**
+     * List all posts in Directus
+     * 
+     * @returns {Promise<Array<PostRecord>>}
+     */
+    static async getAllPosts() {
```

Comment:
As I mention above, we do not have a use-case for this. I.e. there is no reason to retrieve all the posts in one go. As I mention elsewhere, this is very wasteful both in time and bandwidth.